app/user/serializers.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from rest_framework import serializers
+from rest_framework.validators import UniqueValidator
+
+from speaksfer.settings.base import EMAIL_USER
+
+User = get_user_model()
+Profile = get_user_model()
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    username = serializers.CharField(
+        max_length=20,
+        min_length=8,
+    )
+
+    email = serializers.EmailField(
+        required=True,
+        validators=[UniqueValidator(queryset=User.objects.all())],
+    )
+
+    password = serializers.CharField(
+        max_length=128,
+        min_length=8,
+        write_only=True,
+    )
+
+    class Meta:
+        model = User
+        fields = ("email", "username", "password")
+
+    # def validate(self, data):
+
+    #     if data['password'] != data['confirm_password']:
+    #         raise serializers.ValidationError("Passwords do not match")
+    #     return data
+
+    @staticmethod
+    def send_email(user: Any) -> None:
+        email_body = render_to_string(
+            "email_verification.html", {"user": user}
+        )
+        send_mail(
+            "Verify  your email!",
+            email_body,
+            EMAIL_USER,
+            [user.email],
+            fail_silently=False,
+        )
+
+    def create(self, validated_data: Any) -> Any:
+        user = User.objects.create_user(**validated_data)
+        self.send_email(user)
+
+        return user
+
+
+class ProfileSerializer(serializers.ModelSerializer):
+    username = serializers.CharField(source="user.username")
+    bio = serializers.CharField(allow_blank=True, required=False)
+    image = serializers.SerializerMethodField()
```

Comment:
where is the method?

Reply:
Then why do you have the ProfileSerializer?

Reply:
You have not resolved this!

Reply:
The serializer was made when we were working on the user model branch

Reply:
Cause you should not even have the Profile because that belongs to another card